classes/class-p4-metabox-register.php

```diff
@@ -442,4 +414,18 @@ public function cmb2_after_form_do_js_validation( $post_id, $cmb ) {
 	public function is_not_campaign_post() {
 		return P4_Post_Campaign::POST_TYPE !== get_post_type();
 	}
+
+	public static function save_global_project_id( $updated, $action, CMB2_Field $field ) {
+		if ( ! $updated ) {
+			return;
+		}
+		if ( $action === 'removed' ) {
+			update_post_meta( $field->object_id(), 'p4_global_project_tracking_id', null );
+
+			return;
+		}
+
+		$project_id = P4_Analytics_Values::from_cache_or_api_or_hardcoded()->get_id_for_global_project( $field->value() );
+		update_post_meta( $field->object_id(), 'p4_global_project_tracking_id', $project_id );
```

Comment:
The value of `p4_global_project_tracking_id` not saved. 
Tested at https://k8s.p4.greenpeace.org/test-atlas/test-project-id/

Reply:
Ah looks like `get_env` is returning false instead of null here https://github.com/greenpeace/planet4-master-theme/pull/1052/files#diff-51e0c855392eaf41efec35a871fa2dc4R62 , which breaks the null coalescing. Changed it to using `$_ENV` which should do the same thing as `getenv` except it does return null.

Reply:
Hmm something else was going on too, the hidden field actually overwrote what was set here :facepalm: so I removed that field